### PR TITLE
Added Calculated Attributes for CourseUser and fixed Association Name

### PR DESCRIPTION
**Pull request by weiqingtoh**
Smaller PR to #628 - implement methods for CourseUser for User Badge - `experience_points` and `achievement_count`. 

Also fixed the wrong class name in CourseUser association. 

Next step is to implement `user.levels.next`.

### CODE REVIEW

File: app/models/course_user.rb [hunk 1]
```diff
@@ -32,24 +32,34 @@ class CourseUser < ActiveRecord::Base
...
+  calculated :experience_points, (lambda do
+    Course::ExperiencePointsRecord.where do
+      course_experience_points_records.course_user_id == course_users.id
+    end.select { coalesce(sum(course_experience_points_records.points_awarded), 0) }
+  end)
+
+  # @!attribute [r] achievement_count
+  #   Returns the total number of achievements obtained by CourseUser in this course
+  calculated :achievement_count, (lambda do
+    Course::UserAchievement.where { course_user_achievements.course_user_id == course_users.id }.
```
Comment (lowjoel) [inline app/models/course_user.rb]: can we have select then where? lol it reads more naturally that way

File: spec/models/course_user_spec.rb [hunk 2]
```diff
@@ -148,14 +148,39 @@
...
     describe '#experience_points' do
-      let!(:exp_record_1) { create(:course_experience_points_record) }
-      let!(:exp_record_2) do
-        create(:course_experience_points_record, course_user: exp_record_1.course_user)
-      end
-      subject { exp_record_1.course_user }
-      it 'sums all associated experience points records' do
-        points_awarded = exp_record_1.points_awarded + exp_record_2.points_awarded
-        expect(subject.experience_points).to eq(points_awarded)
+      context 'with no experience points record' do
```
Comment (lowjoel) [inline spec/models/course_user_spec.rb]: when there are no experience points records

File: spec/models/course_user_spec.rb [hunk 3]
```diff
@@ -148,14 +148,39 @@
...
+        it 'returns zero' do
+          expect(student.experience_points).to eq(0)
+        end
+      end
+
+      context 'with one or more experience points records' do
```
Comment (lowjoel) [inline spec/models/course_user_spec.rb]: when there are one or more...

**Comment by weiqingtoh** [comment]
@lowjoel updated!

**Pull request merged**